### ISSUE TITLE
Localized strings not appearing when set to Simplified Chinese

### ISSUE THREAD

**Issue by hboon** [AlphaWallet/alpha-wallet-ios#3651]
Verify if simulator only.

1. Change language in Settings.app to Simplified Chinese
2. Launch AlphaWallet app
3. Activity tab

Dates are in Chinese but "Wallet" tab title is still in English. Our localization effort is somewhat patchy and incomplete so not all English strings are localized, but it seems like none of the localized ones are used.

Let's figure out why this is broken before fixing it. It might well be good to stay this way until we patch up the missing localization strings.

<img width="487" alt="Screenshot 2022-01-07 at 10 32 28 AM" src="https://user-images.githubusercontent.com/56189/148481960-342f7664-2f99-4cca-ab18-77e54cc8ddc1.png">

**Comment by eviltofu** [comment]
It appears that all the internationalised strings have to be called. 
```
(lldb) po NSLocalizedString("walletTokens.tabbar.item.title", comment: "")
"Wallet"

(lldb) po Locale.current
▿ zh-Hans_US (current)
  - identifier : "zh-Hans_US"
  - kind : "current"

(lldb) po Locale.current.languageCode
▿ Optional<String>
  - some : "zh"

(lldb) po R.string.localizable.walletTokensTabbarItemTitle(preferredLanguages: ["zh"])
"Wallet"

(lldb) po R.string.localizable.walletTokensTabbarItemTitle(preferredLanguages: ["zh-Hans"])
"我的钱包"

(lldb) po R.string.localizable.walletTokensTabbarItemTitle(preferredLanguages: [])
"Wallet"

(lldb) po Locale.preferredLanguages
▿ 2 elements
  - 0 : "zh-Hans-US"
  - 1 : "en"

(lldb) po Bundle.main.preferredLocalizations
▿ 1 element
  - 0 : "zh-Hans"
```

**Comment by eviltofu** [comment]
Using regular expressions, I was able to convert all the R.string. localizable.xxx to R.string. localizable.xxx(preferredLanguages: nil)

Regular expression:
`R\.string\.localizable.([a-zA-Z0-0]+)(\()(\))`
Replaced by:
`R.string.localizable.$1$2preferredLanguages: nil$3`

There were some instances where I used nil when I first started on this project. Then I'll convert the `nil` to `Bundle.main.preferredLocalizations`

**Comment by hboon** [comment]
> It appears that all the internationalised strings have to be called with the preferredLanguages parameter. I think ALL our localization will not appear.

@eviltofu Can you dig further? Go further back in our repo history, if it still builds. Say around the early in the year 2020. Unless it's changed recently, `R` never requires us to pass in the language to use.

In `257a5308c2617b0aaca1de92ff51311c531f886c` (Oct 2021), R was upgraded, maybe that and the commit before that is a good place to start with.

**Comment by eviltofu** [comment]
Ok I created a new project, localised the strings files, switch to Simplified Chinese, and it works just like you said. 
```
    override func viewDidLoad() {
        super.viewDidLoad()
        titleLabel.text = R.string.file.aTitle()
    }
```
English: `"a.title"="Me";`
Chinese (Simplified): `"a.title"="我";`
Japanese: `"a.title"="私";`
![image](https://user-images.githubusercontent.com/1050309/149050847-781eaae6-206a-4d56-b567-5ab6838a9294.png)

**Comment by eviltofu** [comment]
x`

> > It appears that all the internationalised strings have to be called with the preferredLanguages parameter. I think ALL our localization will not appear.
> 
> @eviltofu Can you dig further? Go further back in our repo history, if it still builds. Say around the early in the year 2020. Unless it's changed recently, `R` never requires us to pass in the language to use.
> 
> In `257a5308c2617b0aaca1de92ff51311c531f886c` (Oct 2021), R was upgraded, maybe that and the commit before that is a good place to start with.

I am not able to build. Xcode is failing to compile several (6 instances) Pods.

**Comment by hboon** [comment]
Some suggestions:

* try manually "bisect" it and see if you can find a commit where it works (and still builds for you)
* strip down the project of most of the code, the extensions and reproduce?
* look at the commit logs messages and see there's a deliberate change to languages/locale that *unintentionally* broke this. For example I see that in 304f5d1ec8b93c5df3ca5240b5c88df32e62b165, we hide the in-app language switcher. Could that PR have broken it?

**Comment by hboon** [comment]
BTW, if it doesn't build on the simulator, try build for the device.

**Comment by eviltofu** [comment]
The App Store version doesn't appear to display simplified Chinese.

**Comment by eviltofu** [comment]
I went back to (129c7b14866e5060930745385f6457fc91eefd54) 31/3/21 when the app doesn't compile anymore. All the versions I compiled never displayed Simplified Chinese.

**Comment by hboon** [comment]
Ok. Do what you can and buzz me if you want me to check too.

**Comment by eviltofu** [comment]
> Ok. Do what you can and buzz me if you want me to check too.

The temporary fix is #3680. I'm puzzled because the new project file I created displays the Simplified Chinese text perfectly!
